src/main/java/org/fenixedu/academic/thesis/ui/service/ThesisProposalsService.java

```diff
@@ -455,6 +459,55 @@ public void accept(StudentThesisCandidacy studentThesisCandidacy) {
         }
     }
 
+    private void createThesisForStudent(StudentThesisCandidacy studentThesisCandidacy) {
```

Comment:
I saw this code somewhere already. You probably need to make the other just delegate here